File: 0201_0300/0231/0231.py
import textwrap
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = f.readline().strip()
    n = 0
    i = 0
    a = ""
    for x in s:
        if x in "0123456789":
            n = n * 10 + "0123456789".index(x)
        elif n == 0:
            a += x
        else:
            a += x * n
            n = 0
    q.write('\n'.join(textwrap.wrap(a, 40)))
                

File: 0201_0300/0231/0231.c
#include <stdio.h>
#include <ctype.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int c, n = 0, i = 0;

	while (isalnum(c = getc(f)))
	{
		if (isdigit(c))
			n = n * 10 + c - '0';
		else if (n == 0)
		{
			putc(c, q);
			if (!(++i % 40))
				putc('\n', q);
		}
		else for (; n > 0; --n)
		{
			putc(c, q);
			if (!(++i % 40))
				putc('\n', q);
		}
	}
	return 0;
}
